### PR TITLE
Add HISTORY section for all functions

### PR DESCRIPTION
**Pull request by baentsch**
This PR
- completes/makes available HISTORY documentation sections for all public functions in libssl & libcrypto that have been added after 3.0.0.
- adds test logic to ascertain this in CI & local testing via enhancements to `util/find-doc-nits` also going forward
- add explanation stating (for contributions of new public functions) to fulfill the user need for documentation showing which OpenSSL version a specific new function first became available
- builds on https://github.com/openssl/openssl/pull/26158 and should cleanly merge after that landed (already approved)

### CODE REVIEW

**Comment by baentsch** [comment]
Ouch -- looks like a huge chunk of new functions without such documentation landed in the rebase... Will add a fix.

File: util/find-doc-nits [hunk 1]
```diff
@@ -862,6 +889,12 @@ sub loadnum ($;$) {
         next if /^#/;
         next if /\bNOEXIST\b/;
         my @fields = split();
+        if ($type && ($type eq "crypto" || $type eq "ssl")) {
+            # 3rd field is version
+            if (not $fields[2] eq "3_0_0") {
+                $history{$fields[0].'(3)'} = $type.$fields[2];
```
Comment (t8m) [inline util/find-doc-nits:895]: This will be broken once we switch to 4.0.0. But I assume we can fix that once we do that.

**Review by t8m — APPROVED** [review]
Approved if CI passes

File: doc/man3/ASYNC_start_job.pod [hunk 2]
```diff
@@ -363,6 +363,8 @@ ASYNC_init_thread, ASYNC_cleanup_thread,
 ASYNC_start_job, ASYNC_pause_job, ASYNC_get_current_job, ASYNC_get_wait_ctx(),
 ASYNC_block_pause(), ASYNC_unblock_pause() and ASYNC_is_capable() were first
 added in OpenSSL 1.1.0.
+ASYNC_set_mem_functions(), ASYNC_get_mem_functions() were added
+in OpenSSL 3.2.0.
```
Comment (mattcaswell) [inline doc/man3/ASYNC_start_job.pod:367]: We are inconsistent about whether we include the patch number or not in the version (i.e. 3.2 or 3.2.0). I think in general we should be using the number *without* the patch, i.e. 3.2. The only time I think we should have an explicit patch number is if we have introduced a function after the initial ".0" release (which is rare).

Comment (baentsch) [reply doc/man3/ASYNC_start_job.pod]: If this is a general statement about the documentation, then that is true (just checked `grep -r "3\.2\.0" doc/ | wc` getting 32 hits). For now the latest commit removed all "3.x.0" instances (8 of the 32 in this example) introduced by this PR _not_ touching the other instances. Please let me know if this is not what you'd expect @mattcaswell .

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]

File: util/find-doc-nits (hunk 1 above)
Comment (baentsch) [reply util/find-doc-nits:895]: Why? Is the idea to not document new functions arriving in 4.0 as having done that (then)?

Comment (baentsch) [reply util/find-doc-nits:895]: Ouch. Interesting to know. So basically all functions still not having a HISTORY entry at that time would get flagged. But whether that's an issue can indeed be decided then. This PR will at least ensure it's only functions added before or at 3.0.0, right?

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**